app/models/Service.ts

```diff
@@ -94,19 +95,30 @@ export const generateServiceDetails = (
     .map((row) => ({ title: row[0], value: row[1] }));
 
 // Determine if a service has its own schedule, or should inherit
-export const shouldServiceInheritScheduleFromOrg = (service: Service) =>
+export const shouldServiceInheritScheduleFromOrg = (service: any) =>
   service.schedule && service.schedule.schedule_days.length > 0;
 
-/**
- * Return a Promise with the fetched Service.
- *
- * Also perform a transformation from the raw API representation of schedules
- * into a nicer-to-use data model of RecurringSchedules.
- */
-export const fetchService = (id: string): Promise<Service> =>
-  get(`/api/v2/services/${id}`).then(({ service }: { service: Service }) => {
-    const recurringSchedule = shouldServiceInheritScheduleFromOrg(service)
-      ? parseAPISchedule(service.schedule)
-      : parseAPISchedule(service.resource.schedule);
-    return { ...service, recurringSchedule };
-  });
+export function fetchServiceSuccessHandler({ service }: { service: any }) {
+  if (shouldServiceInheritScheduleFromOrg(service) && service.schedule) {
+    const recurringSchedule = parseAPISchedule(service.schedule as Schedule);
+    return { ...service, recurringSchedule } as Service;
+  }
+
+  if (service.resource?.schedule) {
+    const recurringSchedule = parseAPISchedule(
+      service.resource?.schedule as Schedule
+    );
+    return { ...service, recurringSchedule } as Service;
+  }
+
+  return UNKNOWN_ERROR_MESSAGE;
+}
```

Comment:
General observation: the amount of ceremony we conduct on the client side to deal with these schedules is a lot. I wonder if there's a simpler way... 🤔